storm-core/pom.xml

```diff
@@ -287,13 +287,13 @@
             <artifactId>metrics-clojure</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.mockito</groupId>
```

Comment:
hamcrest-core dependency is shaded inside the mockito-all and junit brings in another version of hamcrest-core which are incompatible. Excluding hamcrest-core from junit didn't work. This reordering worked. hamcrest-core dependency is needed by ThrowableNestedCauseMapper class.